sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java

```diff
@@ -1601,7 +1602,7 @@ private void showDisabledToast() {
     synchronized void storeDownloadRequestId(DownloadManager downloadManager, DownloadTask task, long downloadId, long enqueueTime) {
 
         /* Check for if state changed and task not canceled in time. */
-        if (mDownloadTask == task) {
+        if (mDownloadTask == task && mReleaseDetails != null) {
```

Comment:
I'd like to remove that check to prove it was the only reason of failing. Also will ask the customer to confirm he's calling notify. Otherwise we would need to root cause again.

Reply:
Confirmed that it's unrelated to user confirmation: https://github.com/microsoft/AppCenter-SDK-Android/issues/1141#issuecomment-491973260

We really need to understand what is going on as hiding a crash will make in-app update silently fail and we'll be really unable to debug it later.

Reply:
So the other scenario is click download, click back and restart activity and click postpone. If async task pool is saturated by other app tasks, you have time to do it. It also explains why it's rare.

Reviewed all code paths and cannot find anything else so this is probably the right one.